src/features/search/components/VnSearchForm/VnSearchForm.tsx

```diff
@@ -0,0 +1,216 @@
+import type { FC } from 'react';
+import { memo, useCallback, useState } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+import { VnSortField, VN_SORT_FIELDS } from 'src/api/models/queryOptions/vn/vnSortField';
+import { VnDevelopmentStatus, VN_DEV_STATUSES } from 'src/api/models/vn/developmentStatus';
+import { VnLength, VN_LENGTHS } from 'src/api/models/vn/length';
+import { ButtonGroup } from 'src/components/ButtonGroup/ButtonGroup';
+import { ControlWrapper } from 'src/components/controls/ControlWrapper';
+import { TextInput } from 'src/components/controls/TextInput';
+import { Field } from 'src/components/Field/Field';
+import { Icon } from 'src/components/Icon/Icon';
+import { IconButton } from 'src/components/IconButton/IconButton';
+import { LanguageSelect } from 'src/components/LanguageSelect/LanguageSelect';
+import { PlatformSelect } from 'src/components/PlatformSelect/PlatformSelect';
+import { Select } from 'src/components/Select';
+import { Slider } from 'src/components/Slider/Slider';
+import { useDebounce } from 'src/hooks/useDebounce';
+import { useTagsQuery } from '../../queries/tag';
+import { VnSearchPopover } from '../VnSearchPopover/VnSearchPopover';
+import type { VnSearchFormValues } from './vnSearchFormValues';
+
+/** Search form component for vns. */
+const VnSearchFormComponent: FC = () => {
+  const [tagsInputValue, setTagsInputValue] = useState('');
+  const debouncedTagInputValue = useDebounce(tagsInputValue);
+
+  const { control } = useFormContext<VnSearchFormValues>();
+  const {
+    fetchNextPage: fetchMoreTags,
+    data: tags,
+    isRefetching: isRefetchingTags,
+    isFetching: isFetchingTags,
+  } = useTagsQuery(debouncedTagInputValue);
+
+  const handleFetchMoreTags = useCallback(() => {
+    if (tags?.pages.at(-1)?.hasMore) {
+      fetchMoreTags();
+    }
+  }, [tags]);
+
+  const handleTagInputChange = useCallback((value: string) => {
+    setTagsInputValue(value);
+  }, []);
+
+  const tagOptions = tags?.pages
+    .flatMap(page => page.results)
+    .map(tag => ({ label: tag.name, value: tag.id })) ?? [];
+
+  const sortFieldOptions = VN_SORT_FIELDS
+    .map(field => ({ value: field, label: VnSortField.toReadable(field) }));
+
+  const devStatusOptions = VN_DEV_STATUSES
+    .map(status => ({ value: status, label: VnDevelopmentStatus.toReadable(status) }));
+
+  const lengthOptions = VN_LENGTHS
+    .map(length => ({ value: length, label: VnLength.toReadable(length) }));
```

Comment:
You can move this stuff out of the component's scope, since it does not rely on props/state